### PR TITLE
🌱Add SpokeBeforeMutation() to FuzzTest to accept data loss from spoke to hub

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/4655

### CODE REVIEW

File: util/conversion/conversion.go [hunk 1]
```diff
@@ -164,6 +164,7 @@ type FuzzTestFuncInput struct {
 	HubAfterMutation func(conversion.Hub)
 
 	Spoke                      conversion.Convertible
+	SpokeBeforeMutation        func(convertible conversion.Convertible)
```
Comment (sbueringer) [inline util/conversion/conversion.go:167]: How will this be set?

**Comment by fabriziopandini** [comment]
/hold
I have commented on the issue about how we are handling this for some conversions.
Let's see if the existing method covers this use cases too

File: util/conversion/conversion.go (hunk 1 above)
Comment (sbueringer) [reply util/conversion/conversion.go:167]: Ah, saw the linked PR now: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2425/files

**Comment by sedefsavas** [comment]
The issue this PR is trying to solve already has a solution described in the issue, closing.

**Pull request closed without merging**